src/pepr/operator/controllers/config/config.ts

```diff
@@ -69,43 +106,126 @@ export async function updateUDSConfig(config: kind.Secret) {
-    UDSConfig.kubeApiCidr = decodedConfigData.KUBEAPI_CIDR;
+  if (networking?.kubeapiCIDR !== UDSConfig.kubeApiCidr) {
+    UDSConfig.kubeApiCidr = networking?.kubeapiCIDR;
     // This re-runs the "init" function to update netpols if necessary
-    log.debug("Updating KubeAPI network policies based on change to kubeApiCidr");
+    configLog.debug("Updating KubeAPI network policies based on change to kubeApiCidr");
     await initAPIServerCIDR();
   }
 
-  // Handle changes to the kubeNodeCidrs
-  if (decodedConfigData.KUBENODE_CIDRS !== UDSConfig.kubeNodeCidrs) {
-    UDSConfig.kubeNodeCidrs = decodedConfigData.KUBENODE_CIDRS;
+  if (!areKubeNodeCidrsEqual(networking?.kubenodeCIDRS, UDSConfig.kubeNodeCidrs)) {
+    UDSConfig.kubeNodeCidrs = networking?.kubenodeCIDRS || [];
     // This re-runs the "init" function to update netpols if necessary
-    log.debug("Updating KubeNodes network policies based on change to kubeNodeCidrs");
+    configLog.debug("Updating KubeNodes network policies based on change to kubeNodeCidrs");
     await initAllNodesTarget();
   }
 
-  if (
-    decodedConfigData.UDS_DOMAIN !== UDSConfig.domain ||
-    decodedConfigData.UDS_ADMIN_DOMAIN !== UDSConfig.adminDomain
-  ) {
-    UDSConfig.domain = decodedConfigData.UDS_DOMAIN;
-    UDSConfig.adminDomain = decodedConfigData.UDS_ADMIN_DOMAIN;
-    if (!UDSConfig.domain || UDSConfig.domain === "###ZARF_VAR_DOMAIN###") {
+  if (expose.domain !== UDSConfig.domain || expose.adminDomain !== UDSConfig.adminDomain) {
+    if (expose.domain && expose.domain !== "###ZARF_VAR_DOMAIN###") {
+      UDSConfig.domain = expose.domain;
+    } else {
       UDSConfig.domain = "uds.dev";
     }
-    if (!UDSConfig.adminDomain || UDSConfig.adminDomain === "###ZARF_VAR_ADMIN_DOMAIN###") {
+    if (expose.adminDomain && expose.adminDomain !== "###ZARF_VAR_ADMIN_DOMAIN###") {
+      UDSConfig.adminDomain = expose.adminDomain;
+    } else {
       UDSConfig.adminDomain = `admin.${UDSConfig.domain}`;
     }
     // todo: Add logic to handle domain changes and update across virtualservices, authservice config, etc
   }
 
   // Update other config values (no need for special handling)
-  UDSConfig.allowAllNSExemptions = decodedConfigData.UDS_ALLOW_ALL_NS_EXEMPTIONS === "true";
+  UDSConfig.allowAllNSExemptions = policy.allowAllNsExemptions === true;
+
+  configLog.info("Updated UDS Config based on uds-operator-config ClusterConfig changes");
+}
 
-  log.info("Updated UDS Config based on uds-operator-config secret changes");
+export async function loadUDSConfig() {
+  // Run in Admission and Watcher pods
+  if (process.env.PEPR_WATCH_MODE || process.env.PEPR_MODE === "dev") {
+    const cfgList = await K8s(ClusterConfig).InNamespace("pepr-system").Get();
+    const cfgSecretList = await K8s(kind.Secret).InNamespace("pepr-system").Get();
+
+    const cfgSecret = cfgSecretList.items.find(s => s.metadata?.name === "uds-operator-config");
```

Comment:
I might be missing intent here but `.Get()` supports getting a secret by name, i.e. this could just be one line:
```suggestion
    const cfgSecret = await K8s(kind.Secret).InNamespace("pepr-system").Get("uds-operator-config");
```

This also could be used for the cluster config, but I see that you're also using the full get to error if multiple cluster configs are found so maybe that one makes sense as is?

Reply:
mmm. ok i wondered if i could do that but the intellisense wasn't indicating `.Get()` accepted params. It's difficult to say whether the # check for the ClucterConfig is valuable here or not. For cluster setups, probably not. On pod cycling or upgrades it makes more sense. I like making the calls the same though .... :shrug: